static/js/commentIcons.js

```diff
@@ -1,179 +1,313 @@
+    self._highlightTargetTextOf(commentId);
+    self._placeCaretAtBeginningOfTextOf(commentId);
     api.triggerCommentActivation(commentId);
   });
 }
 
-exports.showSMHiddenIfTextMarkIsOnIt = function(textMarkId) {
-  var textMarkIsOnSceneMarkHidden = isTextMarkAppliedOnlyOnASMHidden(textMarkId);
 
-  if (textMarkIsOnSceneMarkHidden) {
-    var rep = this.rep;
-    var editorInfo = this.editorInfo;
-    var attributeManager = this.documentAttributeManager;
-    var firstLineWhereTextMarkIsApplied = getFirstLineWhereTextMarkIsApplied(textMarkId, rep);
-    sceneMarkVisibility.showSceneMarksAroundLine(firstLineWhereTextMarkIsApplied, editorInfo, attributeManager);
+commentIcons.prototype._getOrCreateIconsContainerAt = function(top) {
+  var iconContainer = utils.getPadOuter().find('#commentIcons');
+  var iconClass = "icon-at-" + top;
+
+  // is this the 1st comment on that line?
+  var iconsAtLine = iconContainer.find("." + iconClass);
+  var isFirstIconAtLine = iconsAtLine.length === 0;
+
+  // create container for icons at target line, if it does not exist yet
+  if (isFirstIconAtLine) {
+    iconContainer.append('<div class="comment-icon-line ' + iconClass + '"></div>');
+    iconsAtLine = iconContainer.find("." + iconClass);
+    iconsAtLine.css("top", top + "px");
   }
+
+  return iconsAtLine;
 }
 
-var isTextMarkAppliedOnlyOnASMHidden = function(textMarkId) {
-  var $linesWhereTextMarkIsApplied = utils.getPadInner().find('div').has('.' + textMarkId);
-  var textMarkIsAppliedOnlyOnSceneMark = _.every($linesWhereTextMarkIsApplied, function(line){
-    return smUtils.checkIfHasSceneMark($(line));
-  });
-  return textMarkIsAppliedOnlyOnSceneMark && linesAreHidden($linesWhereTextMarkIsApplied);
+commentIcons.prototype._addIcon = function(commentId) {
+  // only create icon if it was not created before
+  var $icon = utils.getPadOuter().find('#icon-' + commentId);
+  if ($icon.length > 0) return;
+
+  // only create icon if commented text was not removed from pad
+  var $inlineComment = utils.getPadInner().find('.comment.' + commentId);
+  if ($inlineComment.length === 0) return;
+
+  var top = $inlineComment.get(0).offsetTop + 2;
+  var iconsAtLine = this._getOrCreateIconsContainerAt(top);
+  var icon = $('#commentIconTemplate').tmpl({ commentId: commentId });
+
+  icon.appendTo(iconsAtLine);
 }
 
-var linesAreHidden = function($lines) {
-  return _.every($lines, function(line){
-    var isLineVisible = line.getBoundingClientRect().height;
-    return !isLineVisible;
+// Update which comments have reply
+commentIcons.prototype._updateCommentIconsStyle = function() {
+  var commentDataManager = pad.plugins.ep_comments_page.commentHandler.commentDataManager;
+  var $iconsContainer = utils.getPadOuter().find('#commentIcons');
+  var $commentsOnText = utils.getPadInner().find('.comment');
+
+  $commentsOnText.each(function() {
+    var classCommentId = /(?:^| )(c-[A-Za-z0-9]*)/.exec($(this).attr('class'));
```

Comment:
se quiser aqui dá para usar o `shared.getCommentIdsFrom($(this).attr('class'))` -- para não ficarmos espalhando regex por aí...